crates/solvers/src/domain/solver/baseline.rs

```diff
@@ -15,35 +15,64 @@ use {
             order::{self, UserOrder},
             solution,
         },
+        infra::config,
     },
     ethereum_types::U256,
-    std::{cmp, collections::HashSet},
+    std::{cmp, collections::HashSet, sync::Arc},
 };
 
-pub struct Baseline {
-    pub weth: eth::WethAddress,
+pub struct Baseline(Arc<Inner>);
+
+struct Inner {
+    weth: eth::WethAddress,
+
     /// Set of tokens to additionally consider as intermediary hops when
     /// path-finding. This allows paths of the kind `TOKEN1 -> WETH -> TOKEN2`
     /// to be considered.
-    pub base_tokens: HashSet<eth::TokenAddress>,
+    base_tokens: HashSet<eth::TokenAddress>,
+
     /// Maximum number of hops that can be considered in a trading path. A hop
     /// is an intermediary token within a trading path. For example:
     /// - A value of 0 indicates that only a direct trade is allowed: `A -> B`
     /// - A value of 1 indicates that a single intermediary token can appear
     ///   within a trading path: `A -> B -> C`
     /// - A value of 2 indicates: `A -> B -> C -> D`
     /// - etc.
-    pub max_hops: usize,
+    max_hops: usize,
+
     /// The maximum number of attempts to solve a partially fillable order.
     /// Basically we continuously halve the amount to execute until we find a
     /// valid solution or exceed this count.
-    pub max_partial_attempts: usize,
+    max_partial_attempts: usize,
 }
 
 impl Baseline {
+    /// Creates a new baseline solver for the specified configuration.
+    pub fn new(config: config::baseline::Config) -> Self {
```

Comment:
Strictly speaking the solver in `domain` should not use a config type from `infra`, right?
Seems like `Baseline` should define what kind of configuration it expects and the `infra` module should return that.
I think this doesn't have to be addressed in this PR. Just noticed it.

Reply:
Good point.

I think this is an issue with all `domain::solver::*::new` methods that take `infra::config` types. At least it is consistent (but different to what we do in the `driver` module.

Reply:
https://github.com/cowprotocol/services/issues/1853

tracked.